medusa/server/api/v2/episode_history.py

```
@@ -0,0 +1,140 @@
+    #: identifier
+    identifier = ('episode_slug', r'[\w-]+')
+    #: path param
+    path_param = ('path_param', r'\w+')
+    #: allowed HTTP methods
+    allowed_methods = ('GET', 'DELETE',)
+
+    def get(self, series_slug, episode_slug, path_param):
+        """Query episode's history information.
+
+        :param series_slug: series slug. E.g.: tvdb1234
+        :param episode_slug: episode slug. E.g.: s01e01
+        :param path_param:
+        """
+        series_identifier = SeriesIdentifier.from_slug(series_slug)
+        if not series_identifier:
+            return self._bad_request('Invalid series slug')
+
+        series = Series.find_by_identifier(series_identifier)
+        if not series:
+            return self._not_found('Series not found')
+
+        if not episode_slug:
+            return self._bad_request('Invalid episode slug')
+
+        episode_number = EpisodeNumber.from_slug(episode_slug)
+        if not episode_number:
+            return self._not_found('Invalid episode number')
+
+        episode = Episode.find_by_series_and_episode(series, episode_number)
+        if not episode:
+            return self._not_found('Episode not found')
+
+        sql_base = '''
+            SELECT rowid, date, action, quality,
+                   provider, version, resource, size, proper_tags,
+                   indexer_id, showid, season, episode, manually_searched
+            FROM history
+            WHERE showid = ? AND indexer_id = ? AND season = ? AND episode = ?
+        '''
+
+        params = [series.series_id, series.indexer, episode.season, episode.episode]
+
+        sql_base += ' ORDER BY date DESC'
+        results = db.DBConnection().select(sql_base, params)
+
+        def data_generator():
+            """Read history data and normalize key/value pairs."""
+            for item in results:
+                d = {}
+                d['id'] = item['rowid']
+
+                if item['indexer_id'] and item['showid']:
```

Can these really be missing?

---

For an episode it can't.
But i've been overthinking the decistion to create the episode_history.py route. (`apiv2/history/tvdb12345/episode/s01e01`). But maybe I should just use a parameter to specify an episode?

---

I think it's consistent with the rest of the routes. But since they can't be missing for episode, we should raise an exception or leave the check out completely?